contentcuration/contentcuration/frontend/shared/views/channel/__tests__/channelDetailsModal.spec.js

```diff
@@ -1,99 +1,236 @@
-import { mount } from '@vue/test-utils';
+import { render, screen, waitFor } from '@testing-library/vue';
+import userEvent from '@testing-library/user-event';
+import { createLocalVue } from '@vue/test-utils';
+import Vuex, { Store } from 'vuex';
 import VueRouter from 'vue-router';
-import ChannelDetailsModal from './../ChannelDetailsModal';
-import storeFactory from 'shared/vuex/baseStore';
-
-const PARENTROUTE = 'Parent route';
-const TESTROUTE = 'test channel details modal route';
-const router = new VueRouter({
-  routes: [
-    {
-      name: PARENTROUTE,
-      path: '/',
-      props: true,
-      children: [
-        {
-          name: TESTROUTE,
-          path: '/testroute',
-          props: true,
-          component: ChannelDetailsModal,
+import ChannelDetailsModal from '../ChannelDetailsModal.vue';
+
+const localVue = createLocalVue();
+localVue.use(Vuex);
+localVue.use(VueRouter);
+
+const channelId = '11111111111111111111111111111111';
+const testChannel = {
+  id: channelId,
+  name: 'Test Channel',
+  description: 'Test Description',
+};
+
+const testDetails = {
+  count: 10,
+  size: 1024,
+};
+
+const mockActions = {
+  loadChannel: jest.fn(() => Promise.resolve(testChannel)),
+  loadChannelDetails: jest.fn(() => Promise.resolve(testDetails)),
+};
+
+const createMockStore = () => {
+  return new Store({
+    state: {
+      connection: {
+        online: true,
+      },
+    },
+    modules: {
+      channel: {
+        namespaced: true,
+        state: {
+          channelsMap: {
+            [channelId]: testChannel,
+          },
+        },
+        getters: {
+          getChannel: state => id => state.channelsMap[id],
         },
-      ],
+        actions: mockActions,
+      },
     },
-  ],
-});
+  });
+};
 
-const store = storeFactory();
-const channelId = '11111111111111111111111111111111';
+const PARENT_ROUTE = 'parent-route';
+const TEST_ROUTE = 'channel-details';
 
-function makeWrapper() {
-  return mount(ChannelDetailsModal, {
-    router,
+const createRouter = () => {
```

Comment:
Can  be simplified to

```
const createRouter = () => {
  return new VueRouter({
    routes: [
      {
        path: '/',
        children: [
          {
            path: '/channel/:channelId',
          },
        ],
      },
    ],
  });
};
```

and you can also remove `router.push` below on line 78.